core/commands/storage/upload/helper/hosts_helper.go

```diff
@@ -101,6 +108,12 @@ func (p *HostsProvider) NextValidHost(price int64) (string, error) {
 	needHigherPrice := false
 LOOP:
 	for true {
+		select {
+		case <-p.ctx.Done():
+			p.cancel()
+			return "", errors.New(failMsg)
+		default:
+		}
```

Comment:
L121: if host is in the blacklist, it should `continue LOOP`? `break LOOP` would quick everything..

Reply:
thank you, it was a stupid bug and i didn't test the repair case.